File: python/examples/calculation.py
"""Demonstrate solar position calculations for Springfield, IL on March 21 at solar noon."""

from datetime import datetime
from zoneinfo import ZoneInfo

from solar_tracker.angles import (
    dual_axis_angles,
    optimal_fixed_tilt,
    single_axis_tilt,
    solar_position,
)


def main():
    latitude = 39.8
    longitude = -89.6

    dt = datetime(2026, 3, 21, 12, 0, tzinfo=ZoneInfo("America/Chicago"))

    pos = solar_position(latitude, longitude, dt)
    sa = single_axis_tilt(pos, latitude)
    da = dual_axis_angles(pos)
    fixed_annual = optimal_fixed_tilt(latitude)

    print("=== Solar Position Calculation Example ===")
    print(f"Location: Springfield, IL ({latitude:.1f}°N, {-longitude:.1f}°W)")
    print(f"Date/Time: {dt}")
    print()
    print("--- Solar Position ---")
    print(f"Day of year: {pos.day_of_year}")
    print(f"Declination: {pos.declination:.2f}°")
    print(f"Equation of Time: {pos.equation_of_time:.2f} minutes")
    print(f"Local Solar Time: {pos.local_solar_time:.2f} hours")
    print(f"Hour Angle: {pos.hour_angle:.2f}°")
    print(f"Zenith Angle: {pos.zenith:.2f}°")
    print(f"Altitude: {pos.altitude:.2f}°")
    print(f"Azimuth: {pos.azimuth:.2f}° (0°=N, 90°=E, 180°=S)")
    print()
    print("--- Optimal Panel Angles ---")
    print(f"Single-axis tracker rotation: {sa:.2f}°")
    print(f"Dual-axis tilt: {da.tilt:.2f}°")
    print(f"Dual-axis panel azimuth: {da.panel_azimuth:.2f}°")
    print(f"Fixed annual optimal tilt: {fixed_annual:.1f}°")


if __name__ == "__main__":
    main()

File: rust/examples/calculation.rs
use chrono_tz::America::Chicago;
use chrono::TimeZone;

use solar_tracker::angles::{
    dual_axis_angles, optimal_fixed_tilt, single_axis_tilt, solar_position,
};

fn main() {
    let latitude = 39.8;
    let longitude = -89.6;

    let dt = Chicago.with_ymd_and_hms(2026, 3, 21, 12, 0, 0).unwrap();

    let pos = solar_position(latitude, longitude, &dt);
    let sa = single_axis_tilt(&pos, latitude);
    let da = dual_axis_angles(&pos);
    let fixed_annual = optimal_fixed_tilt(latitude);

    println!("=== Solar Position Calculation Example ===");
    println!(
        "Location: Springfield, IL ({:.1}°N, {:.1}°W)",
        latitude, -longitude
    );
    println!("Date/Time: {}", dt);
    println!();
    println!("--- Solar Position ---");
    println!("Day of year: {}", pos.day_of_year);
    println!("Declination: {:.2}°", pos.declination);
    println!("Equation of Time: {:.2} minutes", pos.equation_of_time);
    println!("Local Solar Time: {:.2} hours", pos.local_solar_time);
    println!("Hour Angle: {:.2}°", pos.hour_angle);
    println!("Zenith Angle: {:.2}°", pos.zenith);
    println!("Altitude: {:.2}°", pos.altitude);
    println!(
        "Azimuth: {:.2}° (0°=N, 90°=E, 180°=S)",
        pos.azimuth
    );
    println!();
    println!("--- Optimal Panel Angles ---");
    println!("Single-axis tracker rotation: {:.2}°", sa);
    println!("Dual-axis tilt: {:.2}°", da.tilt);
    println!("Dual-axis panel azimuth: {:.2}°", da.panel_azimuth);
    println!("Fixed annual optimal tilt: {:.1}°", fixed_annual);
}
